api/src/pcapi/core/users/email/update.py

```diff
@@ -132,6 +132,29 @@ def confirm_email_update_request(token: str) -> None:
         )
 
 
+def cancel_email_update_request(token: str) -> None:
+    """Cancel the email update request for the given user"""
+
+    payload = ChangeEmailTokenContent.from_token(token)
+    current_email = payload.current_email
+    new_email = payload.new_email
+    user = users_repository.find_user_by_email(current_email)
+    if not user:
+        raise exceptions.InvalidEmailError()
+    check_and_expire_token(user, token, TokenType.CONFIRMATION)
+    try:
+        with transaction():
+            models.UserEmailHistory.build_cancellation(user, new_email)
+            api.suspend_account(
+                user, constants.SuspensionReason.FRAUD_HACK, user, "Suspension suite à un changement d'email annulé"
+            )
+            transactional_mails.send_email_update_cancellation_email(user)
+    except Exception as error:
+        raise ApiErrors(
+            errors={"message": f"erreur inattendue: {error}"},
+        )
+
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏J'aime pas trop catcher tout pour lever une exception floue. 
Ici la seule chose que ton try:except apporte c'est de convertir une 500 en 4xxx 

Je pense que le front ne va pas gérer cette exception différement d'une 500 (je me trompe peut être ici)

<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏

- Soit on ne catche pas
- Soit on catch plus précis avec des messages qui ont du sens